src/execution_plan/ops/op_node_by_label_scan.c

```diff
@@ -90,19 +98,17 @@ static OpResult NodeByLabelScanInit(OpBase *opBase) {
 		return OP_OK;
 	}
 
-	// If we have no children, we can build the iterator now.
-	GraphContext *gc = QueryCtx_GetGraphCtx();
-	Schema *schema = GraphContext_GetSchema(gc, op->n.label, SCHEMA_NODE);
-	if(!schema) {
+	// resolve label ID at runtime
+	_update_label_id(op);
```

Comment:
might be redundant, we're already calling this function at construction time,
I think one of these calls is redundant. please check

Reply:
So we won't forget, writing here the example we started talking about:
`CREATE (:A) RETURN 1 UNION MATCH (a:A) RETURN a`